internal/stringutil/rand_source.go

```diff
@@ -0,0 +1,44 @@
+package stringutil
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var globalRandSource rand.Source
+
+func init() {
+	globalRandSource = NewLockedSource(time.Now().UTC().UnixNano())
+}
+
+// NewLockedSource creates a source of randomness using the given seed.
+func NewLockedSource(seed int64) rand.Source64 {
+	return &lockedSource{
+		lock: new(sync.Mutex),
+		s:    rand.NewSource(seed).(rand.Source64),
+	}
+}
+
+type lockedSource struct {
+	lock sync.Locker
```

Comment:
I don't see any reason to use the interface here, just use Mutex directly.

Reply:
thanks for your comment. i have never used anything other than sync.Locker because i like its simple interface, and the discipline that comes with. in general i also prefer interfaces versus specific implementations if they represent the problem space well. similarly, i have never used `sync.Mutex` directly because i dislike its `TryLock` method, which i find to be unnecessary in all the applications i have ever had. in particular, i see no use for `TryLock` here either. but having said that, i don't have any objections if you personally wanted to change line 24 from `lock sync.Locker` to `lock *sync.Mutex` in the definition of `lockedSource`, please go right ahead. thanks very much again for all your good work on this project.

Reply:
We wouldn't want to use a pointer to the Mutex either.  It's not important, but there are two reasons to avoid an interface here:

1. We are not exposing this to callers, so you may continue to use Mutex the way you are, the interface has no affect on way it is used.
2. The interface adds an additional layer of indirection, which negatively affects the performance of the code, the opposite of what you typically want in a critical section like this.

Fine to leave it as it.